doc/docusaurus/docs/using-plinth/evaluating-plinth.md

```diff
@@ -0,0 +1,201 @@
+result = evaluateCompiledCode compiledCode
+```
+
+The `CodeResult` type is declared as follows:
+
+```haskell
+data CodeResult = CodeResult
+  { codeResult
+      :: Either
+           (CekEvaluationException NamedDeBruijn DefaultUni DefaultFun)
+           (NTerm DefaultUni DefaultFun ())
+  , codeBudget :: ExBudget
+  , codeTraces :: [Text]
+  }
+  deriving stock (Show)
+```
+
+The `codeResult` field contains the result of the evaluation, which can either be a successful evaluation or an error.
+
+The `codeBudget` field contains the execution budget used during the evaluation,
+which includes the CPU and memory usage.
+
+The `codeTraces` field contains the traces emitted during the evaluation.
+
+One can use its `Show` instance to print the result of the evaluation:
+```haskell
+CodeResult
+  { codeResult =
+      Right
+        ( LamAbs
+            ()
+            (NamedDeBruijn{ndbnString = "x", ndbnIndex = 0})
+            ( Apply
+                ()
+                ( Apply
+                    ()
+                    (Builtin () AddInteger)
+                    ( Apply
+                        ()
+                        ( Apply
+                            ()
+                            ( Force
+                                ()
+                                (Builtin () Trace)
+                            )
+                            ( Constant
+                                ()
+                                (Some (ValueOf DefaultUniString "Evaluating x"))
+                            )
+                        )
+                        (Var () (NamedDeBruijn{ndbnString = "x", ndbnIndex = 1}))
+                    )
+                )
+                ( Apply
+                    ()
+                    ( Apply
+                        ()
+                        (Force () (Builtin () Trace))
+                        (Constant () (Some (ValueOf DefaultUniString "Evaluating constant")))
+                    )
+                    (Constant () (Some (ValueOf DefaultUniInteger 2)))
+                )
+            )
+        )
+  , codeBudget =
+      ExBudget
+        { exBudgetCPU = ExCPU 16100
+        , exBudgetMemory = ExMemory 200
+        }
+  , codeTraces = []
+  }
+```
+
+The output is quite verbose and not very readable.
+To make it more readable, we can use the `prettyCodeResult` function, 
+which formats the output in a prettier way:
+
+```
+Evaluation was SUCCESSFUL, result is:
+  \x ->
+    addInteger
+      (force trace "Evaluating x" x)
+      (force trace "Evaluating constant" 2)
+
+Execution budget spent:
+  CPU 16,100
+  MEM 200
+
+No traces were emitted
+```
+
+This output reveals that the evaluation was successful, and the resultng UPLC 
+value is an (un-applied) lambda abstraction. 
+
+The [Lifting Values into CompiledCode](./lifting.md) section contains and example 
+of applying the `CompiledCode` to an argument.
+
+Alternatively, we can use the imported `applyLifted` function like this:
```

Comment:
This part makes it sound like arguments must be a lifted value, but that's not true. You can compile the function using `$$(compile..)`, and compile the argument also using `$$(compile..)`, and then apply the function to the argument using `unsafeApplyCode`. Lifting isn't necessary.

Reply:
You're right that its not always necessary: this is an optional convenience function if one wants to apply and lift in one hop. 
I don't have strong feelings about it, just wanted to come up with the most concise way to apply lifted value: we have 
```haskell
`unsafeApplyCode` liftCodeDef ...
```
in many places...